src/main/kotlin/uk/gov/justice/digital/hmpps/hmppsresettlementpassportapi/service/resettlementassessmentstrategies/AssessmentUtils.kt

```diff
@@ -92,26 +93,49 @@ internal fun ResettlementAssessmentQuestion.removeNestedQuestions() = Resettleme
   detailsContent = this.detailsContent,
 )
 
-internal fun getProfileTag(questionId: String, answer: Answer<*>, pathway: Pathway): List<String> {
+internal fun getProfileTag(answer: Answer<*>, options: List<AssessmentConfigOption>): List<String> {
   val tagFound = mutableListOf<String>()
-  TagAndQuestionMapping.entries.forEach {
-    if ((it.questionId.contains(questionId)) &&
-      (answer.answer.toString().contains(it.optionId)) &&
-      (pathway.name == it.pathway.name)
-    ) {
-      tagFound.add(it.name)
+
+  options.forEach { opt ->
+    log.info("Option Id ${opt.id} and answer id ${answer.answer}")
+    if (answer.answer.toString().contains(opt.id)) {
+      opt.tag?.let { tagFound.add(it) }
     }
   }
+  log.info("Tag Found is ${tagFound.size}")
   return tagFound.distinct()
 }
 
-internal fun processProfileTags(resettlementAssessmentEntity: ResettlementAssessmentEntity, pathway: Pathway): List<String> {
+internal fun processProfileTags(resettlementAssessmentEntity: ResettlementAssessmentEntity, pages: List<AssessmentConfigPage>): List<String> {
   val tagList = mutableListOf<String>()
   resettlementAssessmentEntity.assessment.assessment.forEach {
-    val tagFound = getProfileTag(it.questionId, it.answer, pathway)
+    log.info("question id ${it.questionId}")
```

Comment:
This is a lot of info logs

Reply:
log removed had this for debugging at the time of development